pkg/endpoint/endpoint.go

```diff
@@ -2341,9 +2341,16 @@ func (e *Endpoint) identityLabelsChanged(owner Owner, myChangeRev int) error {
 	e.Mutex.Unlock()
 
 	if readyToRegenerate {
-		e.Regenerate(owner, "updated security labels")
+		status := e.Regenerate(owner, "updated security labels")
+		go func(endpointBuildStatus <-chan bool, e *Endpoint) {
```

Comment:
Why are we only doing this in `identityLabelsChanged()`?

Reply:
In each place we handle it different: 
https://github.com/cilium/cilium/blob/63f3676dfbce11a4d80d209246c0dcfe30906ad7/pkg/endpoint/endpoint.go#L1603-L1625

https://github.com/cilium/cilium/blob/63f3676dfbce11a4d80d209246c0dcfe30906ad7/pkg/endpoint/endpoint.go#L1768-L1774